### ISSUE TITLE
fetchSchemas() does not report stream errors

### ISSUE THREAD

**Issue by kumar303** [mozilla/addons-linter#1252]
In passing, I noticed that the promise returned by `fetchSchemas()` of `src/schema/firefox-schemas-import.js` is not rejected when there is a stream error. It is [resolved here](https://github.com/mozilla/addons-linter/pull/1210/files#diff-44cfe69567cbcc640cec70b9fde0adebR378) but it should probably reject the promise like this:

````js
request.get(downloadUrl(version))
  .pipe(...)
  .on('error', (error) => {
    reject(error);
  })
  .on('end', () => {
    resolve();
  });
````

...otherwise any errors may go unreported which would be confusing.

**Comment by muffinresearch** [comment]
This one should be fixed as part of #1278

**Comment by caitmuenster** [comment]
@muffinresearch Does the pull request cover this issue? Not sure if this is duped or not.
